frontend/src/container/WorkspaceContent.jsx

```diff
@@ -83,45 +88,118 @@ class WorkspaceContent extends React.Component {
-        console.log('%c<WorkspaceContent> Custom event', 'color: #28a745', type, data, state.workspaceIdInUrl)
+    const newUrlSearch = {
+      ...qs.parse(props.location.search),
+      folder_open: [...folderListInUrl, ...contentFolderPath].join(',')
+    }
+
+    props.history.push(PAGE.WORKSPACE.CONTENT_LIST(state.workspaceIdInUrl) + '?' + qs.stringify(newUrlSearch, { encode: false }))
+    this.setState({ appOpenedType: false })
 
-        const contentFolderPath = props.workspaceContentList.filter(c => c.isOpen).map(c => c.id)
-        const folderListInUrl = this.getFolderIdToOpenInUrl(props.location.search)
+    this.setHeadTitle(this.getFilterName(qs.parse(props.location.search).type))
+    this.props.dispatch(resetBreadcrumbsAppFeature())
+  }
 
-        const newUrlSearch = {
-          ...qs.parse(props.location.search),
-          folder_open: [...folderListInUrl, ...contentFolderPath].join(',')
+  handleAllAppChangeLanguage = () => {
+    this.buildBreadcrumbs()
+    if (!this.state.appOpenedType) this.setHeadTitle(this.getFilterName(qs.parse(this.props.location.search).type))
+  }
+
+  // LiveMessage handlers
+  handleWorkspaceModified = data => {
+    if (this.props.currentWorkspace.id !== data.workspace.workspace_id) return
+    this.props.dispatch(setWorkspaceDetail(data.workspace))
+    this.setHeadTitle()
+  }
+
+  handleContentCreated = data => {
+    if (this.props.currentWorkspace.id !== data.workspace.workspace_id) return
+    if (data.content.content_namespace === 'upload') {
+      this.props.dispatch(addWorkspaceShareFolderContentList([
+        {
+          ...data.content,
+          ...(data.content.content_type === CONTENT_TYPE.FOLDER && { parent_id: SHARE_FOLDER_ID })
         }
+      ]))
+    } else {
+      this.props.dispatch(addWorkspaceContentList([data.content]))
+    }
+  }
 
-        props.history.push(PAGE.WORKSPACE.CONTENT_LIST(state.workspaceIdInUrl) + '?' + qs.stringify(newUrlSearch, { encode: false }))
-        this.setState({ appOpenedType: false })
+  handleContentCreatedComment = data => {
+    if (this.props.currentWorkspace.id !== data.workspace.workspace_id) return
+    this.props.dispatch(removeWorkspaceReadStatus(data.content.parent_id))
+  }
 
-        this.setHeadTitle(this.getFilterName(qs.parse(props.location.search).type))
-        this.props.dispatch(resetBreadcrumbsAppFeature())
-        break
-      }
-      case CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE:
-        this.buildBreadcrumbs()
-        if (!state.appOpenedType) this.setHeadTitle(this.getFilterName(qs.parse(props.location.search).type))
-        break
+  handleContentModified = data => {
+    if (this.props.currentWorkspace.id !== data.workspace.workspace_id) return
+    if (data.content.content_namespace === 'upload') {
+      this.props.dispatch(updateWorkspaceShareFolderContentList([
+        {
+          ...data.content,
+          ...(data.content.content_type === CONTENT_TYPE.FOLDER && { parent_id: SHARE_FOLDER_ID })
```

Comment:
while this works, it is too technical because it uses a very advanced knowledge of the spread operator behavior

try to write it in a simpler way